### PR TITLE
poetry documentation

### PR DESCRIPTION
**Pull request by linglp**
Update documentation related to installing poetry.

### CODE REVIEW

**Review by BrunoGrandePhD — CHANGES_REQUESTED** [review]
I made one suggested change. I also think instructions for using Poetry might be better suited for `CONTRIBUTING.md` since they're aimed for developers, but that can be done as part of a larger PR to revamp the docs.

File: README.md [hunk 1]
```diff
@@ -60,6 +60,8 @@ Please note we have a [code of conduct](CODE_OF_CONDUCT.md), please follow it in
 git clone https://github.com/Sage-Bionetworks/schematic.git
 ```
 2. Follow the [instructions](https://python-poetry.org/docs/) here to install `poetry`
+> Note: We recommend installing poetry via `pipx`. (See detail instruction [here](https://github.com/pypa/pipx#:~:text=Other%20Tools.-,Install%20pipx,-On%20macOS))Then use pipx to installl poetry. Your poetry version should be >= 1.2.0. If you have previously installed poetry, try follow the instructions [here](https://python-poetry.org/docs/#:~:text=methods%20above.-,Uninstall%20Poetry,-If%20you%20decide) to unistall it before installing a new version (Or you could try updating poetry by following the instructions [here](https://python-poetry.org/docs/#:~:text=ready%20to%20use!-,Update%20Poetry,-Poetry%20is%20able))
```
[the remark this answers is not in the document]
Comment (linglp) [reply README.md]: @BrunoGrandePhD thanks for the suggestion. I have made the change based on what you recommended.

**Comment by linglp** [comment]
> 



> I made one suggested change. I also think instructions for using Poetry might be better suited for `CONTRIBUTING.md` since they're aimed for developers, but that can be done as part of a larger PR to revamp the docs.

We no longer use `CONTRIBUTING.md`. All the documentation (except the one related to CLI) got consolidated in this README.md (so that users don't have to look different places for documentation)

**Review by andrewelamb — APPROVED** [review]

**Review by BrunoGrandePhD — APPROVED** [review]
By convention, the README and CONTRIBUTING docs have different audiences. README is intended for users, whereas CONTRIBUTING is intended for contributors/developers. In the case of schematic, the installation instructions are distinct for these two audiences. Including instructions for developers in the README can cause confusion for users, especially those who are less technical and don't realize that Poetry is irrelevant to them. 

I'm not going to request that we split up the README in this PR, but I do suggest that this be considered in the future given how schematic users have had consistent trouble with installing schematic. Simplifying the user-facing docs (_i.e._ README) could mitigate these user installation issues. 

I'm approving this PR, but I strongly suggest that you consider my suggested change before merging.

File: README.md [hunk 2]
```diff
@@ -59,7 +59,9 @@ Please note we have a [code of conduct](CODE_OF_CONDUCT.md), please follow it in
 ```
 git clone https://github.com/Sage-Bionetworks/schematic.git
 ```
-2. Follow the [instructions](https://python-poetry.org/docs/) here to install `poetry`
+2. Follow the [instructions](https://python-poetry.org/docs/) here to install `poetry` (version 1.2 or later) using either the official installer or [pipx](https://github.com/pypa/pipx#:~:text=Other%20Tools.-,Install%20pipx,-On%20macOS). If you have an older installation of Poetry, we recommend uninstalling it first. 
+> Note: You could also try updating poetry by following the instructions [here](https://python-poetry.org/docs/#:~:text=ready%20to%20use!-,Update%20Poetry,-Poetry%20is%20able)
```
[the remark this answers is not in the document]
Comment (linglp) [reply README.md]: Thanks Bruno for the update. I was thinking that I could put updating poetry in a separate note (with a different link). But you are right - more links could potentially be confusing.

**Pull request merged**